helm-chart/build.py

```diff
@@ -2,149 +2,193 @@
+    return subprocess.check_output([
+        'git',
+        'log',
+        '-n', '1',
+        '--pretty=format:%cd',
+        '--date=iso',
+        *paths
+    ], **kwargs).decode('utf-8')
 
-def path_changed(paths, commit_range):
-    """Have the path(s) changed during the given commit range?"""
-    if isinstance(paths, str):
-        paths = [paths]
+def path_touched(path, commit_range):
     return subprocess.check_output([
-        'git', 'diff', '--name-only', commit_range, '--',
-    ] + paths).decode('utf-8').strip() != ''
+        'git', 'diff', '--name-only', commit_range, path
+    ]).decode('utf-8').strip() != ''
+
+
+def render_build_args(options, ns):
+    """Get docker build args dict, rendering any templated args."""
+    build_args = options.get('buildArgs', {})
+    for key, value in build_args.items():
+        build_args[key] = value.format(**ns)
+    return build_args
+
+def build_image(image_path, image_spec, build_args):
+    cmd = ['docker', 'build', '-t', image_spec, image_path]
 
+    for k, v in build_args.items():
+        cmd += ['--build-arg', '{}={}'.format(k, v)]
+    subprocess.check_call(cmd)
 
-def build_images(prefix, images, commit_range=None, push=False):
-    for image in images:
-        image_path = os.path.join(BASEPATH, 'images', image)
+def build_images(prefix, images, tag=None, commit_range=None, push=False):
+    value_modifications = {}
+    for name, options in images.items():
+        image_path = os.path.join('images', name)
+        paths = options.get('paths', []) + [image_path]
         if commit_range:
-            if not path_changed(IMAGE_FILES, commit_range):
-                print("Skipping {}, not touched in {}".format(image, commit_range))
+            if not path_touched(*paths, commit_range=commit_range):
+                print("Skipping {}, not touched in {}".format(name, commit_range))
                 continue
-        tag = last_git_modified(IMAGE_FILES)
-        image_spec = '{}{}:{}'.format(prefix, image, tag)
+        last_commit =last_modified_commit(*paths)
+        if tag is None:
+            tag = last_commit
+        image_name = prefix + name
+        image_spec = '{}:{}'.format(image_name, tag)
+
+        template_namespace = {
+            'LAST_COMMIT': last_commit,
+            'TAG': tag,
+        }
+        build_args = render_build_args(options, template_namespace)
+
+        build_image(image_path, image_spec, build_args)
+        value_modifications[options['valuesPath']] = {
+            'name': image_name,
+            'tag': tag
+        }
 
-        subprocess.check_call([
-            'docker', 'build', '-t', image_spec,
-            '--build-arg', 'BINDERHUB_VERSION=%s' % tag,
-            image_path
-        ])
         if push:
             subprocess.check_call([
                 'docker', 'push', image_spec
             ])
+    return value_modifications
+
+def build_values(name, values_mods):
+    rt_yaml = YAML()
```

Comment:
`rt` Return? Something-type? It's not clear to me what the rt is shorthand for. Perhaps mention at first use in a comment.